### DISCUSSION TITLE
_tw and WindPress

### DISCUSSION THREAD

**Discussion by suasgn** [gregsullivan/_tw#137] [Ideas]
Hi Greg,
I wanted to thank you for all this work on _tw.

I'm maintaining a WordPress plugin named [WindPress](https://github.com/wind-press/windpress) that offers Tailwind CSS v3 and v4 integration for WordPress.
The plugin scope is focused on bringing the joy of the Tailwind CSS experience without requiring a development environment that supports modern Javascript build tools like Node.js.

We are looking for a starter theme to make this experience seamless, and we found that _tw shares our vision.

It would be great if we could collaborate on this matter and make _tw the official starter theme for WindPress.

Please let me know if you are interested in this collaboration. I'm looking forward to hearing from you.

**Comment by gregsullivan** [comment]
Hi Joshua—Great to hear from you!

I'm all for there being more ways to use Tailwind with WordPress, and I'm happy to hear you're working on another one. How would you see a collaboration working? My main concern is that _tw is designed to be used alongside a development environment with modern build tools, which seems in conflict with what you're looking to create.

You may have already seen it, but there is a [Why _tw?](https://underscoretw.com/docs/why/) page in the _tw docs where I outline my intentions and goals for the project in a bit more detail.

Looking forward to hearing more!

**Reply by suasgn** [reply]
Hi Greg, an update from us.

We have developed a [modified _tw theme](https://wind.press/blog/underscore-wt) to demonstrate how to integrate WindPress with WordPress themes. This can serve as a reference for other WordPress themes.

I have viewed the [Why _tw?](https://underscoretw.com/docs/why/) page and believe we share the same intentions and goals.

- [x] **WordPress first**
- [x] **Block editor focused** + support adding arbitrary classes in the block editor, and other page builders.
- [x] **Classic editor friendly**
- [x] **Just Tailwind and WordPress**: plug-n-play Tailwind integration via WindPress plugin, and modern development workflow friendly.
- [x] **Free of production dependencies**: A single cached CSS file for all pages & no client-side resources to generate CSS classes (instead, this is done on the admin side).

**Reply by gregsullivan** [reply]
Hello—Thanks for the update!

I think the full context from that page on production dependencies better explains my position:

> There are other projects at the nexus of WordPress and Tailwind that take different approaches to adding Tailwind support, and many of them do so by running Tailwind in production. There are a number of ways to do this, but all require some combination of:
> 
> - Using server- or client-side resources to generate CSS classes
> - Caching CSS classes on a per-page basis, dependent on which classes are used on which pages
> 
> I’m not comfortable with either of the above options. These are necessary compromises if you want to support adding arbitrary classes in the block editor, but I worry about potential failures in class generation, caching, and other related brittleness or slowness inherent in running Tailwind in production.
> 
> And absolutely there are use cases for the above, and if you have them, those projects will hopefully serve you well! In my opinion, though, generating CSS locally is the most robust approach, and one that ensures the best experience for website visitors.

My concern with generating in the admin comes from the added complexity in either trying to get all of the Tailwind classes used across the site's posts and pages when generating a single file, or in creating separate files for each page, combined with my belief that developers are generally better served not using Tailwind classes at the database level.

All of that said, I don't want this to come off as too negative! I will reemphasize this part of the quote above:

> And absolutely there are use cases for the above, and if you have them, those projects will hopefully serve you well!

If someone wants to add arbitrary Tailwind classes in the block editor, your project makes more sense than mine. For the client sites I work on, I would never want my clients to be using Tailwind themselves, and I prefer to have all of the styles for the pages I create be part of the theme itself. (I know this isn't a universal view, though!)

Best of luck with your project!
